projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts

```diff
@@ -24,50 +24,146 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
+    *```html
+    *<igx-tabs-group label="Tab 1" icon="folder">
+    *```
+    */
     @Input()
     public label: string;
 
+    /**
+    * An @Input property that sets the value of the `label`.
+    *```html
+    *<igx-tabs-group label="Tab 1" icon="home">
+    *```
+    */
     @Input()
     public icon: string;
 
+    /**
+    * An @Input property that allows you to enable/disable the `isDisabled`.
+    *```html
+    *<igx-tabs-group label="Tab 2  Lorem ipsum dolor sit" icon="home" isDisabled="true">
+    *```
+    */
     @Input()
     public isDisabled: boolean;
 
-    @HostBinding('attr.role') public role = 'tabpanel';
-
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.role')
+    public role = 'tabpanel';
+
+    /**
+     * An accessor that returns the class of the tab.
+     * ```html
+     * @ViewChild("MyTab")
+     * public tab: IgxTabsGroupComponent;
+     * ngAfterViewInIt(){
+     *    let tabStyleClass = this.tab.styleClass;
+     *    this.cdr.detectChanges();
+     * }
+     * ```
+     * Disclaimer:
+     * Svetla Boykova will fix the code to make sense.
+     */
     @HostBinding('class.igx-tabs__group')
     get styleClass(): boolean {
         return true;
     }
 
+    /**
+     * An accessor that returns the value of the `_itemStyle` property.
+     * ```html
+     * @ViewChild("MyTab")
+     * public tab: IgxTabsGroupComponent;
+     * ngAfterViewInIt(){
+     *    let tabStyle = this.tab.itemStyle;
+     *    this.cdr.detectChanges();
+     * }
+     * ```
+     */
     public get itemStyle(): string {
         return this._itemStyle;
     }
 
+    /**
+     * An accessor that returns the `tab` component.
+     * ```html
+     * @ViewChild("MyTab")
+     * public tab: IgxTabsGroupComponent;
+     * ngAfterViewInIt(){
+     *    let tabComponent = this.tab.relatedTab;
+     *    this.cdr.detectChanges();
+     * }
+     * ```
+     */
     get relatedTab(): IgxTabItemComponent {
         if (this._tabs.tabs) {
             return this._tabs.tabs.toArray()[this.index];
         }
     }
 
+    /**
+     * An accessor that returns the value of the index of the tab.
+     * ```html
+     * @ViewChild("MyTab")
+     * public tab: IgxTabsGroupComponent;
+     * ngAfterViewInIt(){
+     *    let tabIndex = this.tab.index;
+     *    this.cdr.detectChanges();
+     * }
+     * ```
+     */
     get index() {
         if (this._tabs.groups) {
             return this._tabs.groups.toArray().indexOf(this);
         }
     }
 
+    /**
+     * An accessor that returns the `_tabTemplate` property.
+     * ```html
+     * @ViewChild("MyTab")
+     * public tab: IgxTabsGroupComponent;
+     * ngAfterViewInIt(){
+     *    let tabTemplate = this.tab.customTabTemplate;
+     *    this.cdr.detectChanges();
+     * }
+     * ```
+     */
     get customTabTemplate(): TemplateRef<any> {
         return this._tabTemplate;
     }
 
+    /**
+     * An accessor that modifies the `_tabTemplate` property.
```

Comment:
Maybe it is better that we say the dewvelopers to use this to set a custom template to the tab. They do not need to know about the _tabTemplate property that will hold it.